server/src/main/java/org/opensearch/action/search/TransportSearchAction.java

```diff
@@ -490,11 +428,11 @@ private void executeRequest(
             relativeStartNanos,
             System::nanoTime
         );
-
-        final List<SearchRequestOperationsListener> searchListenersList = createSearchListenerList(originalSearchRequest, timeProvider);
-        SearchRequestContext searchRequestContext = new SearchRequestContext(
-            new SearchRequestOperationsListener.CompositeListener(searchListenersList, logger)
-        );
+        final boolean phaseTookEnabled = clusterService.getClusterSettings().get(SEARCH_PHASE_TOOK_ENABLED)
+            || (originalSearchRequest.isPhaseTook() != null && originalSearchRequest.isPhaseTook());
```

Comment:
This isn't the same logic as before. The existing behavior said that the request-level value (if set) always takes precedence over the cluster-level parameter.

The logic should be:
```
final boolean phaseTookEnabled;
if (originalRequest.isPhaseTook() == null) {
  phaseTookEnabled = clusterService.getClusterSettings().get(SEARCH_PHASE_TOOK_ENABLED);
} else {
  phaseTookEnabled = originalRequest.isPhaseTook();
}
```

Reply:
Good catch! let me update this.